TrapListener: Support authoritative engineID discovery (#394)
gosnmp/gosnmp: CHANGELOG.md

## unreleased

> NOTE: The signature of UnmarshalTrap now returns an additional error: #394

Reviewer: I'm not sure what signature means in this context. I think you meant method return?

```suggestion
NOTE: The UnmarshalTrap now returns both an SnmpPacket and an error (#394)
```

> * [ENHANCEMENT] Support authoritative engineID discovery when listening for traps #394

Reviewer: Since this is breaking, we call this a `CHANGE`.

```suggestion
* [CHANGE] Support authoritative engineID discovery when listening for traps #394
```
